src/CalledMethods.cpp

```diff
@@ -26,9 +27,17 @@ void CalledMethods::leastUpperBound(PVAliasSet &preSet, MethodsSet &curMethodsSe
     preSet.methodsSetIntersection(curMethodsSet);
 }
 
-void CalledMethods::onAnnotation(PVAliasSet *input, std::string &fnName,
-                                 AnnotationType annotationType) {
-    if (annotationType == AnnotationType::CallsAnnotation) {
-        input->addMethod(fnName);
+void CalledMethods::onAnnotation(PVAliasSet* input, Annotation* annotation, const std::string& invokerFnName) {
+    if (annotation->getAnnotationType() == AnnotationType::CallsAnnotation) {
+        auto annoMethods = annotation->getAnnotationMethods();
+        for (std::string annoMethod : annoMethods) {
+            input->addMethod(annoMethod);
+
+        }
+
+        if (ParameterAnnotation *paramAnno = dynamic_cast<ParameterAnnotation *>(annotation)) {
+            input->addMethod(invokerFnName);
```

Comment:
I don't think this is correct: it doesn't match my intuition. This might be why I'm confused about the expected output of a test case, too. Let me explain my thinking:
* when some method is called, e.g., `foo(a, b)`, regardless of the annotations on the parameters of `foo`, I expected `a` and  `b` to have `foo` in their CM sets. This should happen regardless of annotations.
* if there are annotations on the parameters (e.g., suppose the second parameter of `foo` has `Calls("free", "x")`, I expect that those are _also_ applied, but that the two things should happen separately. So, after this call to foo, I'd expect the CM store to look like: `{ a -> CM("foo"), b -> CM("foo"), a.x -> CM("free") }`. However, I think this code results in the following CM store: `{ a -> CM("foo"), b -> CM("foo"), a.x -> CM("free", "foo") }`. Note the extra "foo" in `a.x`'s CM set, which is what I'm concerned about.